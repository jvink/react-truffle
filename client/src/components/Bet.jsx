import React from 'react';
import BetForm from './BetDetails';
import App from '../App'
import '../css/bet.css';
import BetDetailsComponent from './BetDetails';

 const BetComponent = (props) => {
    const { bet, type, changeContent} = props;

    const onClickCallTwoFunctions = () => {
        props.onClickDetail(props.changeContent);
        props.onClickSetBetId(bet.id);
    }
    
    const renderStatus = () => {
        switch (type) {
            case 0:
                return <div className="statusWrapper" style={{ backgroundColor: '#F79F1F' }}>Pending</div>
            case 1:
                return <div className="statusWrapper" style={{ backgroundColor: '#009432' }}>Closed</div>
            case 2:
                return <div className="statusWrapper" style={{ backgroundColor: '#2980b9' }}>Payed</div>
            default:
                return <div className="statusWrapper" style={{ backgroundColor: '#e74c3c' }}>Invalid</div>
        }
    }

    return (
    <div className="betWrapper" onClick={onClickCallTwoFunctions}>
            <span><b>{bet.name}</b></span>
            {renderStatus()}
            <span>Inzet: ${bet.inzet}</span>
            <span>Temperatuur: {bet.guess}°</span>
            <div className="detailButton"><b>></b></div>
            {((type === 1 || type === 2) && bet.guess === bet.winning_degree) &&
                <div className="wonWrapper">
                    <span role="img" aria-labelledby="emoticon">🎊</span><span> You Won! </span><span role="img" aria-labelledby="emoticon">🎉</span>
                </div>
            }
        </div>
    )
}

export default BetComponent;