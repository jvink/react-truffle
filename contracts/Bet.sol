pragma solidity ^0.5.0;

contract Bet{
    // Model a Candidate
    // Store Candidates
    // Fetch Candidate

    struct User {
        uint id;
        string name;
        uint voteCount;
    }

    mapping(uint => User) public users;  // Write to the blockchain (data layers) gonna cost gas
     // Store Candidates Count
    uint public usersCount;

    constructor () public {
        addCandidate("Candidate 1");
        addCandidate("Candidate 2");
    }

    function addCandidate (string memory _name) private {
        usersCount ++;
        users[usersCount] = User(usersCount, _name, 0);
    }
}
