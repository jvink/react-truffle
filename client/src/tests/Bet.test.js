import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-dom/extend-expect';
import Bet from '../components/Bet';

afterEach(cleanup);

it('renders', () => {
    const { asFragment } = render(<Bet type={1} bet={
        {
            cityId: '123',
            name: 'Rotterdam',
            inzet: 12,
            guess: 22,
            time: '2019-06-12:00:00+01:00',
            timeOfNow: '2019-06-06T18:52:26+01:00'
        }
    } />);
    expect(asFragment()).toMatchSnapshot();
});

it('renders correct status', () => {
    const { getByTestId } = render(<Bet type={0} bet={
        {
            cityId: '1234',
            name: 'Dordrecht',
            inzet: 20,
            guess: 15,
            time: '2019-06-13:00:00+01:00',
            timeOfNow: '2019-06-06T18:52:26+01:00'
        }
    } />);

    expect(getByTestId("status")).toHaveTextContent("Pending");
});