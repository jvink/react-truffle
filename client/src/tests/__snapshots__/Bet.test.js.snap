// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="betWrapper"
  >
    <span>
      <b>
        Rotterdam
      </b>
    </span>
    <div
      class="statusWrapper"
      data-testid="status"
      style="background-color: rgb(0, 148, 50);"
    >
      Closed
    </div>
    <span>
      Inzet: $12
    </span>
    <span>
      Temperatuur: 22Â°
    </span>
    <div
      class="detailButton"
    >
      <b>
        &gt;
      </b>
    </div>
  </div>
</DocumentFragment>
`;
